File: 374. Guess Number Higher or Lower/solution.java
/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class Solution extends GuessGame {
  public int guessNumber(int n) {
    int hi = n;
    int lo = 0;
    int mid = 0;
    while(lo <= hi) {
      mid = lo + (hi - lo) / 2;
      int currGuess = guess(mid);
      if (currGuess == 0) return mid;
      if (currGuess == -1) hi = mid - 1;
      if (currGuess == 1) lo = mid + 1;
    }
    return 0;
  }
}

File: 374. Guess Number Higher or Lower/solution.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
  def guessNumber(self, n):
    """
    :type n: int
    :rtype: int
    """
    upper_bound = n
    lower_bound = 1
    while(True):
      cur_guess = (lower_bound + upper_bound) / 2
      res = guess(cur_guess)
      if res == 0: return cur_guess
      elif res == 1: lower_bound = (lower_bound + upper_bound + 1) / 2
      else: upper_bound = (lower_bound + upper_bound - 1) / 2